ui/src/pages/publications/[id]/index.tsx

```diff
@@ -84,12 +84,16 @@ type Props = {
 const Publication: Types.NextPage<Props> = (props): React.ReactElement => {
     const router = useRouter();
     const confirmation = Contexts.useConfirmationModal();
-    const [isBookmarked, setIsBookmarked] = React.useState(props.bookmark ? true : false);
+    const [isBookmarked, setIsBookmarked] = React.useState(props.bookmark);
     const [isPublishing, setPublishing] = React.useState<boolean>(false);
     const [approvalError, setApprovalError] = React.useState('');
     const [serverError, setServerError] = React.useState('');
     const [isEditingAffiliations, setIsEditingAffiliations] = React.useState(false);
 
+    useEffect(() => {
```

Comment:
is this useEffect necessary? since the bookmark is already fetched SSR and it comes via props

Reply:
It's needed to prevent the bookmark icon incorrectly keeping its state when navigating from one publication / topic to another